moviebot/controller/controller_flask.py

```diff
@@ -0,0 +1,398 @@
+        """Sends template, buttons or text based on current agent options.
+
+        Args:
+            user_id: User id.
+            payload: User payload.
+
+        Returns:
+            Object with message to send to the server.
+        """
+        self.continue_dialogue(user_id, payload)
+        if self.user_options[user_id]:
+            if "**" in self.agent_response[user_id]:
+                return self.user_messages[user_id].movie_template(
+                    self.info[user_id], self.agent_intent
+                )
+            else:
+                buttons = self.user_messages[user_id].create_buttons(
+                    self.user_options[user_id]
+                )
+                return self.user_messages[user_id].buttons_template(
+                    buttons,
+                    self.agent_response[user_id],
+                    intent=self.agent_intent,
+                )
+        else:
+            return self.user_messages[user_id].text(
+                self.agent_response[user_id], intent=self.agent_intent
+            )
+
+    def run_method(self, user_id: str, payload: str) -> Union[bool, Callable]:
+        """Runs methods for specific user inputs.
+
+        Args:
+            user_id: User id.
+            payload: User payload.
+
+        Returns:
+            Output of the payload action if possible.
+        """
+        for item in self.methods:
+            if payload.lower() == item["payload"]:
+                func = item.get("action")
+                return func(user_id)
+        return True
+
+    def exit(self, user_id: str) -> None:
+        """Ends conversation and deletes user id from agent.
+
+        Args:
+            user_id: User id.
+        """
+        self.agent_response[
+            user_id
+        ] = "You are exiting. I hope you found a movie. Bye."
+        self.user_messages[user_id].text(
+            self.agent_response[user_id], intent=self.agent_intent
+        )
+        del self.agent[user_id]
+
+    def instructions(self, user_id: str) -> HTTP_OBJECT_MESSAGE:
+        """Creates utterance with usage instructions.
+
+        The instructions are sent when the conversation starts and when '/help'
+        is received.
+
+        Args:
+            user_id: User id.
+
+        Returns:
+            Object with instructions message to send to the server.
+        """
+        response = (
+            "To start the conversation say Hi or Hello, or simply "
+            'enter you preferences ("I want a horror movie from the 90s").\n\n'
+            'To restart the recommendation process, issue "/restart".\n\n'
+            'To end the conversation, issue "/exit" or say Bye/Goodbye.\n\n'
+            'To see these instructions again, issue: "/help".'
+        )
+
+        return self.user_messages[user_id].text(
+            response, intent="REVEAL.DISCLOSE"
+        )
+
+    def store_data(self, user_id: str) -> HTTP_OBJECT_MESSAGE:
```

Comment:
Ambiguous name. Is it soring data?

Reply:
It is not, this is related to the storage policy. The method was renamed.